skel/daemons/settings.go

```diff
@@ -0,0 +1,368 @@
+			createDepositAndHotWallet(peatioClient, currencyGroup, opendaxAddr)
+		}
+	} else {
+		for _, currencyGroup := range currenciesGroups {
+			walletsGroup := findCurrenciesInWallets(wallets, currencyGroup)
+			for walletKey, walletsGroup := range walletsGroup {
+				// If wallet constains partially filled currencies
+				// System will update current wallet with all currencies list
+				// If wallet doesn't constain currencies
+				// System will update create wallet
+				if walletKey == "partial" {
+					for _, w := range walletsGroup {
+						res := getWallet(peatioClient, w)
+						if res != nil {
+							// For example, If there is difference in second slice
+							// It means that second slice is not fully included on first slice
+							// []string{"eth", "link", "usdt"} VS []string{"eth", "link"}
+							// res: []string{"usdt"}, []string{}
+							_, secondDiff := funk.Difference(res.Currencies, currencyGroup)
+							if len(secondDiff.([]string)) > 0 {
+								mergedCurrencies := append(res.Currencies, currencyGroup...)
+								updateWallet(peatioClient, w, funk.UniqString(mergedCurrencies))
+							}
+						}
+					}
+				} else if walletKey == "none" {
+					createDepositAndHotWallet(peatioClient, currencyGroup, opendaxAddr)
+				}
+			}
+		}
+	}
+}
+
+// Method divideCurrenciesIntoGroups divides currencies into groups
+// In first place system always puts parent ID
+// For example {"eth": ["eth", "usdt", "link"]}
+func divideCurrenciesIntoGroups(currencies []CurrencyResponse) map[string][]string {
+	res := make(map[string][]string)
+	// Sort currencies with empty Parent ID first
+	// It means those currencies should have separate wallet
+	sort.SliceStable(currencies, func(i, j int) bool {
+		return currencies[i].ParentID < currencies[j].ParentID
+	})
+
+	for _, currency := range currencies {
+		if currency.Type == "coin" {
+			if currency.ParentID == "" {
+				res[currency.ID] = append(res[currency.ID], currency.ID)
+			} else {
+				res[currency.ParentID] = append(res[currency.ParentID], currency.ID)
+			}
+		}
+	}
+
+	return res
+}
+
+// Method createDepositAndHotWallet creates deposit and hot wallet
+func createDepositAndHotWallet(peatioClient *peatio.Client, currencies []string, opendaxAddr string) {
+	// Create Deposit Wallet
+	depositWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Deposit Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "deposit",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, depositApiError := peatioClient.CreateWallet(depositWalletParams)
+	if depositApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", depositApiError.Error, depositApiError.Errors)
+		return
```

Comment:
we cant return errors here, as it will stop creating other wallets in the loop
Like we will not do anything with this error in the cycle, that's why we just display